applications/visualizer/frontend/src/models/workspace.ts

```diff
@@ -25,16 +26,23 @@ export class Workspace {
   store: ReturnType<typeof configureStore>;
   layoutManager: LayoutManager;
   updateContext: (workspace: Workspace) => void;
-
-  constructor(id: string, name: string, datasetIds: Set<string>, activeNeurons: Set<string>, updateContext: (workspace: Workspace) => void) {
+  globalContext: GlobalContextType;
+  constructor(
+    id: string,
+    name: string,
+    datasetIds: Set<string>,
+    activeNeurons: Set<string>,
+    updateContext: (workspace: Workspace) => void,
+    globalContext: GlobalContextType,
```

Comment:
issue (blocking): I don't think this is a good idea. If we need to know what datasets exist we should pass only the datasets, not the whole context. Possibly we can replace the     `datasetIds: Set<string>`, with `Record<string, Dataset>` and we avoid the _initializeActiveDatasets all along